src/Models.jl

```diff
@@ -37,11 +37,15 @@ Convention is that `weights` defaults to `StatsBase.uweights(Float32, size(outpu
 function fit end
 
 """
-    predict(::Model, input)
+    predict(model::Model, inputs::AbstractMatrix)
 
-Predict targets for the provided `input` and [`Model`](@ref).
+Predict targets for the provided the collection of `inputs` and [`Model`](@ref).
 
-Returns a predictive distribution or point estimates depending on the [`Model`](@ref).
+If the `estimate_type(model)` is [`PointEstimate`](@ref) then this function should return
+another `AbstractMatrix` in which each column contains the prediction for a single input.
+
+If the `estimate_type(model)` is [`DistributionEstimate`](@ref) then this function should
+return a `AbstractVector{<:Distribution}`.
```

Comment:
It's good to specify such unspoken convention. I wonder whether you need to force anything through `output traits`?

> If the `estimate_type(model)` is [`DistributionEstimate`](@ref) then this function should
return a `AbstractVector{<:Distribution}`

If you want to be more strict, the return is probably `AbstractMatrix{<:UnivariateDistribution}` or `AbstractVector{<:MultivariateDistribution}` or `MatrixDistribution`. We only came across `AbstractVector{<:MultivariateDistribution}` in BaselineModels, though

Reply:
Hmm unless you object I think I'm going to stick with this. It seems to be what's getting used, and it seems to me to be reasonably general.

> It's good to specify such unspoken convention. I wonder whether you need to force anything through output traits?

The tests in `Models.TestUtils` already do this, although they don't do it quite right: #18 .